core/src/main/scala/com/nvidia/spark/rapids/tool/analysis/AppSQLPlanAnalyzer.scala

```diff
@@ -339,6 +363,8 @@ class AppSQLPlanAnalyzer(app: AppBase, appIndex: Int) extends AppAnalysisBase(ap
           } else {
             taskUpatesSubset(taskUpatesSubset.size / 2)
           }
+          updateStageDiagnosticMetrics(stageId, accumInfo.infoRef.getName,
+            StatisticsMetrics(min, median, max, sum))
```

Comment:
this goes along with Ahmed's comments above. We are storing the same information twice. here and then in AccumProfileResults.  Can't we just create one StatisticsMetrics and reuse it?

Reply:
thanks @tgravescs! I refactored this by creating one copy of `AccumProfileResults` and reuse in the mapping `stageToDiagnosticMetrics`